File: onem2m-application/nodejs/onem2m-get-all-cin.js
// Version 1.1

var request = require('request');

var sensorDataContainer = "TiltSensor/DATA";

var options = {
	uri: "http://127.0.0.1:8080/server/" + sensorDataContainer + "?rcn=4",
	method: "GET",
	headers: {
		"Content-Type": "application/json"
	}
};

console.log("Sending request >>> ");
console.log(options.method + " " + options.uri);

request(options, function (err, resp, body) {
	console.log("\n\n<<< Response received ! ");
	if(err) {
		console.log("Error = " + err);
	} else {
		console.log("Response Status Code = " + resp.statusCode);
		console.log("Response Body = " + body);

		var obj = JSON.parse(body);
		console.log("\n\nEffective content of CINs = ");
		obj["m2m:cnt"]["cin"].forEach(elt => {
			console.log(elt.con);	
		});
	}
});

File: onem2m-application/python/onem2m-get-all-cin.py
# Version 1.1

import requests
import json

sensorDataContainer = "TiltSensor/DATA"

print("Sending request >>> ")
response = requests.get('http://127.0.0.1:8080/server/' + sensorDataContainer + '?rcn=4', 
                        headers={'Accept': 'application/json'})

print("<<< Response received ! ")

if response.status_code != 200:
    print("Error = ", response.text)
else:
    print("Effective content of CINs = ")
    contentInstanceInJSON = json.loads(response.content)
    for elt in contentInstanceInJSON['m2m:cnt']['cin']:
        print("   " + elt['con'])


